proxy/src/main/java/com/velocitypowered/proxy/protocol/ProtocolUtils.java

```diff
@@ -194,12 +203,95 @@ public static void writeProperties(ByteBuf buf, List<GameProfile.Property> prope
     return properties;
   }
 
+  /**
+   * Reads an byte array for legacy version 1.7
+   *
+   * @param buf the buffer to read from
+   * @return the read byte array
+   */
+  public static byte[] readArray17(ByteBuf buf) {
+    // Read in a 2 or 3 byte number that represents the length of the packet. (3 byte "shorts" for
+    // Forge only)
+    // No vanilla packet should give a 3 byte packet
+    int len = readVarShort(buf);
+
+    // (Integer.MAX_VALUE & 0x1FFF9A ) = 2097050 - Forge's current upper limit
+    Preconditions.checkArgument(len <= (Integer.MAX_VALUE & 0x1FFF9A),
+        "Cannot receive array longer than 2097050 (got %s bytes)", len);
+
+    byte[] ret = new byte[len];
+    buf.readBytes(ret);
+    return ret;
+  }
+
+  // Legacy 1.7
+
+  /**
+   * Legacy.
+   * @param b array
+   * @param buf buf
+   * @param allowExtended forge
+   */
+  public static void writeArray17(byte[] b, ByteBuf buf, boolean allowExtended) {
+    // (Integer.MAX_VALUE & 0x1FFF9A ) = 2097050 - Forge's current upper limit
+    if (allowExtended) {
+      Preconditions
+          .checkArgument(b.length <= (Integer.MAX_VALUE & 0x1FFF9A),
```

Comment:
Put `Integer.MAX_VALUE & 0x1FFF9A` in a constant value, and use it for the message.